src/app/portlets/dot-edit-page/content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.ts

```diff
@@ -26,6 +27,8 @@ export class DotEditPageToolbarComponent implements OnInit, OnChanges {
     lockerModel: boolean;
     mode: PageMode;
 
+    private debounceStateSelector = _.debounce((pageState: PageMode) => this.setSelectorState(pageState), 500, { leading: true });
```

Comment:
500ms? you sure? that looks too slow to wait for an user when click something.

Reply:
on the test I did it looked good to me! what do you think would be a proper ms?

Reply:
Nah... all good I tested it and seems to be working fine.